priv/perf/apps/load_test/lib/ethereum/ethereum.ex

```diff
@@ -108,6 +125,222 @@ defmodule LoadTest.Ethereum do
+      WatcherInfoAPI.Api.Transaction.create_transaction(LoadTest.Connection.WatcherInfo.client(), transaction)
+
+    result = Jason.decode!(response.body)["data"]
+    process_transaction_result(result, amount_in_wei, input_address, output_address, currency, tries)
+  end
+
+  def submit_transaction(typed_data, sign_hash, private_keys) do
+    signatures =
+      Enum.map(private_keys, fn private_key ->
+        sign_hash
+        |> to_binary()
+        |> signature_digest(private_key)
+        |> Encoding.to_hex()
+      end)
+
+    typed_data_signed = Map.put_new(typed_data, "signatures", signatures)
+
+    submit_typed(typed_data_signed)
+  end
+
+  defp to_binary(hex) do
+    hex
+    |> String.replace_prefix("0x", "")
+    |> String.upcase()
+    |> Base.decode16!()
+  end
+
+  defp signature_digest(hash_digest, private_key_binary) do
+    {:ok, <<r::size(256), s::size(256)>>, recovery_id} =
+      :libsecp256k1.ecdsa_sign_compact(
+        hash_digest,
+        private_key_binary,
+        :default,
+        <<>>
+      )
+
+    # EIP-155
+    # See https://github.com/ethereum/EIPs/blob/master/EIPS/eip-155.md
+    base_recovery_id = 27
+    recovery_id = base_recovery_id + recovery_id
+
+    <<r::integer-size(256), s::integer-size(256), recovery_id::integer-size(8)>>
+  end
+
+  defp submit_typed(typed_data_signed, retry_count \\ 120)
+
+  defp submit_typed(typed_data_signed, 0), do: execute_submit_typed(typed_data_signed)
+
+  defp submit_typed(typed_data_signed, counter) do
+    {:ok, response} = execute_submit_typed(typed_data_signed)
+    decoded_response = Jason.decode!(response.body)["data"]
+
+    case decoded_response do
+      %{"messages" => %{"code" => "submit:utxo_not_found"}} ->
+        Process.sleep(1_000)
+        submit_typed(typed_data_signed, counter - 1)
+
+      %{"messages" => %{"code" => "operation:service_unavailable"}} ->
+        Process.sleep(1_000)
+        submit_typed(typed_data_signed, counter - 1)
+
+      %{"txhash" => _} ->
+        decoded_response
+    end
+  end
+
+  defp execute_submit_typed(typed_data_signed) do
+    WatcherInfoAPI.Api.Transaction.submit_typed(LoadTest.Connection.WatcherInfo.client(), typed_data_signed)
+  end
+
+  defp process_transaction_result(result, amount_in_wei, input_address, output_address, currency, tries) do
+    case {result, tries} do
+      {%{"code" => "create:client_error", "messages" => %{"code" => "operation:service_unavailable"}}, 0} ->
+        {:error, result}
+
+      {%{
+         "result" => "complete",
+         "transactions" => [
+           %{
+             "sign_hash" => sign_hash,
+             "typed_data" => typed_data,
+             "txbytes" => txbytes
+           }
+         ]
+       }, _} ->
+        {:ok, [sign_hash, typed_data, txbytes]}
+
+      _ ->
+        Process.sleep(1_000)
+        create_transaction(amount_in_wei, input_address, output_address, currency, tries - 1)
+    end
+  end
+
+  defp root_chain_get_eth_balance(address, 0) do
```

Comment:
Just return error here. Calls are done in the second definition. We can get the same by just increasing counter value by 1 when calling `root_chain_get_eth_balance`.

Reply:
it seems like this conversation was marked as resolved, but not addressed?

Reply:
It was trying to fetch balance. Now I'm raising an exception https://github.com/omgnetwork/elixir-omg/pull/1745/files#diff-3e3ab486b14889eab1712a81574bd2c6a194527da3c65eec103b399e86a6a896R255
Should I return {:error, reason} ?